src/AdminConsole/Endpoints/ApplicationEndpoints.cs

```diff
@@ -33,4 +41,25 @@ public static IEndpointRouteBuilder MapApplicationEndpoints(this IEndpointRouteB
                 statusCode: e.Details.Status);
         }
     }
+
+    public static async Task<IResult> DownloadOnboardingAsync(
+        [FromServices] IApplicationService applicationService,
+        [FromServices] IOptionsSnapshot<PasswordlessManagementOptions> options,
+        [FromRoute] string appId)
+    {
+        var onboarding = await applicationService.GetOnboardingAsync(appId);
+
+        if (onboarding == null)
+        {
+            return NotFound();
+        }
+
+        var contentBuilder = new StringBuilder();
+        contentBuilder.AppendLine($"API Url: {options.Value.ApiUrl}");
+        contentBuilder.AppendLine($"Public ApiKey: {onboarding.ApiKey}");
+        contentBuilder.AppendLine($"Private ApiSecret: {onboarding.ApiSecret}");
+
+        var byteArray = Encoding.UTF8.GetBytes(contentBuilder.ToString());
+        return File(byteArray, "application/octet-stream", $"passwordless-apikeys-{appId}.txt");
```

Comment:
Could be simplified with slightly fewer allocations:

```suggestion
        using var memory = new MemoryStream();
        using var writer = new StreamWriter(memory);
        writer.WriteLine($"API Url: {options.Value.ApiUrl}");
        writer.WriteLine($"Public ApiKey: {onboarding.ApiKey}");
        writer.WriteLine($"Private ApiSecret: {onboarding.ApiSecret}");

        return File(memory.ToArray(), "application/octet-stream", $"passwordless-apikeys-{appId}.txt");
```